src/robotFactory.js

```diff
@@ -1,10 +1,94 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    if (step > 0) {
+      this.coords.y -= step;
+    }
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    if (step > 0) {
+      this.coords.x += step;
+    }
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    if (step > 0) {
+      this.coords.x -= step;
+    }
+
+    return this;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    if (step > 0) {
+      this.coords.z += step;
+    }
+
+    return this;
+  }
+
+  goDown(step = 1) {
+    if (step > 0) {
+      this.coords.z -= step;
+    }
+  }
```

Comment:
```suggestion
    }
    return this;
  }
```
`return this` was not required in the task but you have in other movements so it needs to be consistent.

Reply:
agree with @ysrckr